Jenkinsfile

```diff
@@ -303,393 +307,61 @@ pipeline {
-        }
-      }
-      post {
-        success {
-          setGitHubBuildStatus('platform/package', 'Package', 'SUCCESS')
-        }
-        failure {
-          setGitHubBuildStatus('platform/package', 'Package', 'FAILURE')
-        }
-      }
-    }
-
-    stage('Run "dev" functional tests') {
-      steps {
-        setGitHubBuildStatus('platform/ftests/dev', 'Functional tests - dev environment', 'PENDING')
-        container('maven') {
-          echo """
-          ----------------------------------------
-          Run "dev" functional tests
-          ----------------------------------------"""
-          script {
-            try {
-              runFunctionalTests('nuxeo-distribution/nuxeo-server-tests')
-              runFunctionalTests('nuxeo-distribution/nuxeo-server-hotreload-tests')
-              runFunctionalTests('nuxeo-distribution/nuxeo-server-gatling-tests')
-              runFunctionalTests('ftests')
-              setGitHubBuildStatus('platform/ftests/dev', 'Functional tests - dev environment', 'SUCCESS')
-            } catch (err) {
-              setGitHubBuildStatus('platform/ftests/dev', 'Functional tests - dev environment', 'FAILURE')
+          dir('ci/helm/preview') {
+            echo """
+            ----------------------------------------
+            Deploy Preview environment
+            ----------------------------------------"""
+            // first substitute docker image names and versions
+            sh """
+              mv values.yaml values.yaml.tosubst
+              envsubst < values.yaml.tosubst > values.yaml
+            """
+            // second create target namespace (if doesn't exist) and copy secrets to target namespace
+            script {
+              sh(returnStdout: true, script: 'jx -b ns | sed -r "s/^Using namespace \'([^\']+)\'.+\\$/\\1/"')
+              boolean nsExist = sh(returnStatus: true, script: "kubectl get namespace ${NAMESPACE}") == 0
+              String noCommentOpt = '';
+              if (nsExist) {
+                noCommentOpt = '--no-comment'
+              } else {
+                sh "kubectl create namespace ${NAMESPACE}"
+              }
+              sh "kubectl create secret generic preview-docker-cfg --namespace=${NAMESPACE} --from-file=.dockerconfigjson=/home/jenkins/.docker/config.json --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"
+              // third build and deploy the chart
+              // waiting for https://github.com/jenkins-x/jx/issues/5797 to be fixed in order to remove --source-url
+              sh """
+                jx step helm build --verbose
+                mkdir target && helm template . --output-dir target
+                jx preview \
+                  --namespace ${NAMESPACE} \
+                  --verbose \
+                  --source-url=https://github.com/nuxeo/nuxeo \
+                  --preview-health-timeout 15m \
```

Comment:
This was added in arender because nuxeo 10.10 hotfixes itself during boot. This is not the case with master, so:
- probably not needed
- can be lowered

Reply:
Actually the preview often takes 10-11 minutes before being available:
https://jenkins.platform.dev.nuxeo.com/blue/organizations/jenkins/nuxeo%2Fnuxeo/detail/PR-3856/85/pipeline/